### DISCUSSION TITLE
What in this version's txt2img allows the itterations and samples to be in true sequential order?

### DISCUSSION THREAD

**Discussion by Wonderflex** [basujindal/stable-diffusion#110] [Q&A]
In Basujindal's version if you run a prompt that dictates a seed, and has more than one iteration or sample, the subsequent seeds used are in exact order.

For example:

--prompt "dogs in cars" --seed 1 --n_iter 2 --n_samples 1 y

Yields two images, one that matches running the prompt only on seed one, and the other as if you ran it only on seed 2.

In the default img2txt version the same prompt yields the first image being correctly from seed 1, but the second image is from a random seed, and not seed 2.

I've figured out how to add the seed number like this version has, but it only adds a number to the seed count and doesn't actually step up the seed number by 1 when running it.  So the file name text says it is the next seed, but it is still the same random second seed, unlike Basujindal's that will actually progress one seed forward at a time.  

What in this version is dictating that each iteration should be the next seed number in order?  The default version runs iterations much faster for me, but I really need the seeds to be in order like this version has. 

Thanks

**Comment by bitRAKE** [comment]
The default img2txt only uses the seed at the start to initialize all state; whereas Basujindal's version opt.seed is send to the model.sampler, and opt.seed increments after every image save.

**Comment by chipyards** [comment]
In this version (Basu Jindal), the individual seed of each image is determined in ddpm.py, by the following piece of code which appears more than once:
``torch.manual_seed(seed)
tens.append(torch.randn(img_shape, device=self.cdevice))
seed+=1``
This seems to override the effect of:
``seed_everything(opt.seed)``
which is in optimized_txt2img.py.
In older versions, this single call to seed_everything() causes a new random seed to be generated for each sample.
This behavior reappears with option --fixed_code (in this case the seed information appended to filenames is wrong, this is a (small) bug.)
Note: this is what I observed, but since this code is frequently updated, this may be wrong at the time you are reading!
